source/TS.NET/Drivers/XMDA/Thunderscope.cs

```diff
@@ -392,31 +393,29 @@ private void UpdateAdc()
 
         private void UpdateAfe(int channel, ref ThunderscopeChannel channelConfiguration)
         {
-            CalculatePgaConfigurationWord(ref channelConfiguration);
+            ThunderscopeChannelCalibration channelCal = channel switch
+            {
+                0 => calibration.Channel1,
+                1 => calibration.Channel2,
+                2 => calibration.Channel3,
+                3 => calibration.Channel4,
+            };
+
+            CalculatePgaConfigurationWord(ref channelCal, ref channelConfiguration);
             SetPGA(channel, channelConfiguration.PgaConfigurationWord);
-            Thread.Sleep(10);
+            Thread.Sleep(1);
```

Comment:
Delay should actually be moved to the i2c and spi write functions, spi is pretty much always fine since it's fast, i2c is slow enough that we may send things back to back before the state machine resets

Reply:
If there is a delay micros we can use that, otherwise SPI is running with no delay and all I2C just uses 1milli